astroplan/scheduling.py

```diff
@@ -140,6 +140,76 @@ def create_score_array(self, time_resolution=1*u.minute):
             score_array *= constraint(self.observer, targets, times)
         return score_array
 
+    def weighted_score_array(self, time_resolution=1*u.minute):
+        """
+        For each block, this returns a score for each time using the
+        formula (score*weight+score*weight+...)/(weight+weight+..)
+        """
+        # note: with this method, a non-boolean constraint with weight
+        # zero has the exact same effect if it was boolean
+        start = self.schedule.start_time
+        end = self.schedule.end_time
+        times = time_grid_from_range((start, end), time_resolution)
+        score_array = np.zeros((len(self.blocks), len(times)))
+        zeros = np.ones((len(self.blocks), len(times)), dtype=int)
+        local_constraints = []
+        weights = []
+        for i, block in enumerate(self.blocks):
+            weights.append(0)
+            local_constraints.append([])
+            # schedulers put global constraints into ._all_constraints
+            # so we can use .constraints for the local constraints
+            if block.constraints:
+                for constraint in block.constraints:
+                    local_constraints[i].append(constraint.__class__.__name__)
+                    applied_score = constraint(self.observer, [block.target],
+                                               times=times)[0]
+                    if constraint.boolean_constraint:
+                        # add to the mask designating where the score is zero
+                        zeros[i] &= applied_score
+                    # TODO: make a default weight=1 and merge these
```

Comment:
Yes, but we need to decide if we want weighting to be part of the constraints package (add a kwarg to `Constraint`) or the scheduling package (`constraint_weights` kwarg for `ObservingBlock` and `global_constraint_weights` kwarg for `Schedule`). During the implementation of either method it would be very easy to set the default weight and fix this TODO.